app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,78 @@
+class CategoriesController < ApplicationController
+
+  before_action :require_login
+
+  def index
+    @categories = Category.all
+    @user = User.find(session[:user_id])
```

Comment:
this is exactly what our `current_user` method gives us! so we don't need to write this line in every method - we can just reference `current_user` - both in our controllers AND our views